### ISSUE TITLE
double free or corruption (out)

### ISSUE THREAD

**Issue by arkdae** [sahib/rmlint#626]
I have run rmlint twice now,

`rmlint -g --algorithm=sha3-256 -D -w -x -C`

not on identical directories, but both times it ends with this:

`double free or corruption (out)`

The rmlint.sh file's last line isn't finished, and the program is hung. No Ctrl-C works, I must put it in the background and kill it.

I am running on Arch Linux.

**Comment by Cebtenzzre** [comment]
This is a duplicate of #562, fixed in commit b76f85322109d15536d54c91524de1316b96dff5. Please build from the latest master.

**Comment by arkdae** [comment]
I have updated to the AUR version of rmlint-git. I'm pretty sure it should be the latest:

```
rmlint --version
version 2.10.1 compiled: Jul 12 2023 at [21:03:56] "Ludicrous Lemur" (rev dc923ed3)
compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support

rmlint was written by Christopher <sahib> Pahl and Daniel <SeeSpotRun> Thomas.
The code at https://github.com/sahib/rmlint is licensed under the terms of the GPLv3.
```

And yet I am still getting the same behavior.

**Comment by Cebtenzzre** [comment]
Yes, that is the latest version. I might already have a fix for this somewhere, I'll have to look into this further when I have the time.

**Comment by jim-collier** [comment]
I get this error too, also with the same result of interrupted `json` and `csv` files. 

I see that the main issue is closed #562.

I'm on freshly compiled:

~~~
version 2.10.3 compiled: Jul  7 2025 at [07:50:24] "Ludicrous Lemur" (rev a7e1df4b)
compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support
~~~

Command-line:

`sudo rmlint "/mnt/bfs/bf7" --no-crossdev --see-symlinks --hidden --xattr -o progressbar -o summary --no-with-color --types=none,duplicates,duplicatedirs -o csv:/var/log/rmlint.csv -o json:/var/log/rmlint.json`

My use case is

1. Use `rmlint` to find df and dd.
2. Use that information to manually delete completely, many TBs of unnecessary redundant files and even whole folders, that I don't want around even they were deduplicated. Just their existence as files and folders makes manual organization difficult. This step is necessarily manual, as there's no `rmlint` rules (or any conceivable rules) I could use to consistently base deletion on.
3. For edge cases or where I'm not sure, punt for now and run `rmlint` with df only, and .sh output, to actually deduplicate the remaining duplicate files - to free up another ten TBs or so.

Since the volume is about 50 TB of slow HDDs and with so much duplicate data it takes days for `rmlint` to scan (and 15 drives grinding away), I only want to run the `rmlint` phase that does full file-hashes once (in step 1 above), and write the hashes to xattrs. Then use that info the second time for the actual later duduping step, reading from xattrs without having to rescan.

Writing to xattrs is also important because the files almost never change over years - but move around often, including to different arrays with xattrs and mtime preserved. And deduping every few years is important. So having that sort of makeshift location-agnostic hash 'cache' on so many large files is important.

Edit: Same result as above, with just `--types=none,duplicates` (no `duplicatedirs`).

**Comment by vassilit** [comment]
@arkdae and @jim-collier you might want to retest with the latest mater (or rmlint-git AUR package).
